src/app/core/services/utils/helper.service.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class HelperService {
+
+  public extractId(data: string) {
+    const vimeoRegex = /^.*(vimeo\.com\/)((channels\/[A-z]+\/)|(groups\/[A-z]+\/videos\/))?([0-9]+)/;
+    const ytRegex = /^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#\&\?]*).*/;
+
+    const ytId = data.match(ytRegex);
+    const vimeoId = data.match(vimeoRegex);
+
+    if (ytId) {
```

Comment:
I jak będą dochodzić kolejni providerzy to będziesz kolejne ify dodawał? To już lepiej żeby provider miał swoja metodę do ekstrakcji i tutaj je zbiorczo wykorzystać.